### PR TITLE
Added Squish option to the Transform Text dialog

### PR DESCRIPTION
**Pull request by Gioche6**
Fixes issue #6705 

@africanmathsinitiative/developers , This is ready for review. 

@rdstern, I have added the squish radio button as requested to pass the function `str_squish.` 
Additionally, with the help of @Ivanluv , we have refactored the codes.

### CODE REVIEW

**Comment by lloyddewit** [comment]
@Ivanluv Please could you peer review?
@rdstern Please could you test?
Thanks!

**Comment by Ivanluv** [comment]
@rdstern  could you test this before I can review

**Review by rdstern — APPROVED** [review]
Over to @shadrackkibet or @lloyddewit for second aspproval and merging!

**Comment by shadrackkibet** [comment]
@Ivanluv your code review is still pending? Let me know if I misunderstood.

**Comment by Ivanluv** [comment]
@shadrackkibet Am reviewing it now

File: instat/dlgTransformText.vb [hunk 1]
```diff
@@ -185,9 +197,10 @@ Public Class dlgTransformText
         clsConvertFunction = New RFunction
         clsLengthFunction = New RFunction
         clsPadFunction = New RFunction
-        clsTrimFunction = New RFunction
         clsWordsFunction = New RFunction
         clsSubstringFunction = New RFunction
+        clsSquishTrimFunction = New RFunction
+
```
Comment (Ivanluv) [inline instat/dlgTransformText.vb:203]: could we have this as one line spacing?

File: instat/dlgTransformText.vb [hunk 2]
```diff
@@ -200,6 +213,7 @@ Public Class dlgTransformText
         ucrChkLastOr.Checked = False
 
 
+
```
Comment (Ivanluv) [inline instat/dlgTransformText.vb:216]: One line spacing here also

File: instat/dlgTransformText.vb (hunk 1 above)
Comment (Gioche6) [reply instat/dlgTransformText.vb:203]: @Ivanluv , this is done

File: instat/dlgTransformText.vb (hunk 2 above)
Comment (Gioche6) [reply instat/dlgTransformText.vb]: @Ivanluv , this is done

**Comment by Ivanluv** [comment]
This is okay from my end. Over to you @lloyddewit

**Comment by Gioche6** [comment]
@rdstern , @lloyddewit , This PR is all ready for review and merging.

File: instat/dlgTransformText.vb [hunk 3]
```diff
@@ -330,84 +355,69 @@ Public Class dlgTransformText
...
-        TestOkEnabled()
-        'if rdoWords.Checked change word parametervalues everytime changes are detected on 
-        'ucrReceiverFirstWord, ucrReceiverLastWord, ucrNudFirstWord, ucrNudLastWord
-        If rdoWords.Checked Then
-            changeWordParamValues()
+    Private Sub LastAndFirstWord_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrChkFirstOr.ControlValueChanged, ucrChkLastOr.ControlValueChanged
+        If ucrChangedControl Is ucrChkFirstOr AndAlso ucrChkFirstOr.Checked Then
+            ucrReceiverFirstWord.SetMeAsReceiver()
+        ElseIf ucrChkLastOr.Checked AndAlso ucrChangedControl Is ucrChkFirstOr Then
```
Comment (lloyddewit) [inline instat/dlgTransformText.vb:378]: Is this more readable if we make it consistent with the other conditions?
```suggestion
        ElseIf ucrChangedControl Is ucrChkFirstOr AndAlso ucrChkLastOr.Checked Then
```

File: instat/dlgTransformText.vb [hunk 4]
```diff
@@ -421,4 +431,35 @@ Public Class dlgTransformText
...
+                clsSquishTrimFunction.SetRCommand("str_trim")
+                If rdoLeftPad.Checked Then
+                    clsSquishTrimFunction.AddParameter("side", Chr(34) & "left" & Chr(34), iPosition:=2)
+                ElseIf rdoRightPad.Checked Then
+                    clsSquishTrimFunction.AddParameter("side", Chr(34) & "right" & Chr(34), iPosition:=2)
+                ElseIf rdoBothPad.Checked Then
+                    clsSquishTrimFunction.AddParameter("side", Chr(34) & "both" & Chr(34), iPosition:=2)
+                End If
+            End If
+        End If
```
Comment (lloyddewit) [inline instat/dlgTransformText.vb:462]: Is it more readable if we follow the 'happy path stays left' principle?
```suggestion
        If Not rdoTrim.Checked Then
            Exit Sub
        End If

        If rdoSquish.Checked Then
            clsSquishTrimFunction.SetRCommand("str_squish")
            clsSquishTrimFunction.RemoveParameterByName("side")
        Else
            clsSquishTrimFunction.SetRCommand("str_trim")
            If rdoLeftPad.Checked Then
                clsSquishTrimFunction.AddParameter("side", Chr(34) & "left" & Chr(34), iPosition:=2)
            ElseIf rdoRightPad.Checked Then
                clsSquishTrimFunction.AddParameter("side", Chr(34) & "right" & Chr(34), iPosition:=2)
            ElseIf rdoBothPad.Checked Then
                clsSquishTrimFunction.AddParameter("side", Chr(34) & "both" & Chr(34), iPosition:=2)
            End If
        End If
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@Gioche6 Thanks, you implemented this very cleanly and it looks good.
I just had a couple of small suggestions.
We can then merge.

**Comment by Gioche6** [comment]
@lloyddewit , Thankyou for the thoughtful suggestion. I have gone through them and I feel they are great. I have also been implemented now.

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern since your last approval, there are just a couple of small changes that should not have affected the functionality. Please could you approve again? We can then merge.
Thanks

**Review by rdstern — APPROVED** [review]

**Pull request merged**